__tests__/app.test.js

```diff
@@ -205,7 +204,75 @@ describe("Articles Endpoint", () => {
         .get("/api/articles/notAnId/comments")
         .expect(400)
         .then(({ body }) => {
-          expect(body.msg).toBe("Bad request: Invalid category type");
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+  });
+  describe("POST:/api/articles/:article_id/comments", () => {
+    test("POST:201 - Responds with an array containing correctly formated comment object", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({
+          username: "butter_bridge",
+          body: "This is a test comment",
+        })
+        .expect(201)
+        .then(({ body }) => {
+          const comment = body.comment[0];
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            author: "butter_bridge",
+            created_at: expect.any(String),
+            votes: 0,
+            body: "This is a test comment",
+            article_id: 1,
+          });
+        });
+    });
+    test("POST:400 - Responds with an error when attempting to POST request with a body that does not contain the correct fields", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({})
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad request");
+        });
+    });
+    test("POST:400 - Responds with an error when attempting to make a POST request with valid fields but the value of a field is invalid", () => {
+      // Comment for NC staff members: Since PSQL is doing type coercion even if i send a non-string this test will fail without the use of an external library. Also, the other way i could check is converting the string values i get from req.body to another type in an if statement like if(Number(usermame)) then reject there but I would have to check mutiple data types. Please suggest another way if you can but for now i will use a library called express-validator I will only use it for this case.
```

Comment:
This is probably not worth testing for. Could a comment just be numbers? Probably... and you can be pretty confident it's not going to break anything too. Unlike the other way around.